test-basic/optic-fromDocUris.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const should = require('should');
+
+const marklogic = require('../');
+const p = marklogic.planBuilder;
+
+const pbb = require('./plan-builder-base');
+var testconfig = require('../etc/test-config.js');
+const execPlan = pbb.execPlan;
+const getResults = pbb.getResults;
+var db = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+var q = marklogic.queryBuilder;
+const Stream = require('stream');
+let result = new Set();
+
+describe('fromDocUris', function() {
+    this.timeout(15000);
```

Comment:
@mariusmarklogic  @anu3990  Any time there's a timeout/sleep in a test, we should have a comment to explain why. We want tests to run as quickly as possible, so we only want a timeout if we really need it - and then we need that comment to explain why we need it.

Reply:
Hi Rob, this is a common timeout for all the tests in this file. This is primarily for Jenkins. Mocha has a timeout of 2 seconds. On jenkins we use mocha to run the tests. Due to the sync/async path Node follows, it sometimes takes more than 2 seconds to complete. So this is a way to override the mocha auto-timeout.